### PR TITLE
enh(docs) improve API docs

### LINKED ISSUE #1576: Documentation is confusing
http://highlightjs.readthedocs.io/en/latest/api.html#highlight-name-value-ignore-illegals-continuation

Listed arguments don't match with the signature. It's very confusing and I have to read the code to understand what the arguments are.

Edit: I just realized I confused input arguments with return value. It should be more clear. I suggest putting the "Returns an object..." sentence in a separate paragraph, and naming the first argument "language" or "language_name", the second "html_string", etc.

### PR DESCRIPTION
**Pull request by yyyc514**
Closes #1576.

This doesn't change any API, just renames a few arguments to be clearer.

### CODE REVIEW

File: src/highlight.js [hunk 1]
```diff
@@ -513,16 +513,24 @@ https://highlightjs.org/
...
+   * @returns an object that represents the result
+   * @property {string} language - the language name
+   * @property {number} relevance - the relevance score
+   * @property {string} value - the highlighted HTML code
+   * @property {mode} top - top of the current mode stack
+   * @property {boolean} illegal - indicates whether any illegal matches were found
   */
-  function highlight(name, value, ignore_illegals, continuation) {
+  function highlight(languageName, code, ignore_illegals, continuation) {
+    var value = code;
```
Comment (egor-rogov) [inline src/highlight.js]: Is there any reason not to replace `value` with `code`?

Comment (yyyc514) [reply src/highlight.js]: Wasn’t in the mood for a search and replace battle and value is so generic and the scope was large. Lol.

**Comment by yyyc514** [comment]
@egor-rogov I renamed value but left a longer name to make this easier to change in the future.  Right now our scope is too large IMHO... otherwise this wouldn't be an issue.

File: src/highlight.js [hunk 2]
```diff
@@ -513,16 +513,24 @@ https://highlightjs.org/
...
+   * @returns an object that represents the result
+   * @property {string} language - the language name
+   * @property {number} relevance - the relevance score
+   * @property {string} value - the highlighted HTML code
+   * @property {mode} top - top of the current mode stack
+   * @property {boolean} illegal - indicates whether any illegal matches were found
   */
-  function highlight(name, value, ignore_illegals, continuation) {
+  function highlight(languageName, code, ignore_illegals, continuation) {
+    var codeToHighlight = code; // for search and replace
```
Comment (egor-rogov) [inline src/highlight.js]: `codeToHighlight` is fine, but again, why not simply
```js
function highlight(languageName, codeToHighlight, ignore_illegals, continuation) {
```
?
Doesn't renaming by assigning the value to another variable look strange?

Comment (yyyc514) [reply src/highlight.js]: I see what you're getting at, but doesn't bother me. In the method definition all that wording is extraneous... highlight(codeToHighlight) is silly and repetitive.  But in the huge scope this variable is used in the longer naming makes a lot more sense.

So this is the best of both worlds IMHO.

If you insist I'll change it, but I'd rather not. :-)

Comment (egor-rogov) [reply src/highlight.js]: Okay, agree. But let's remove `// for search and replace`?

Comment (yyyc514) [reply src/highlight.js]: LOL, I thought that would appease you... I'll remove.

Comment (egor-rogov) [reply src/highlight.js]: I think it's rather more confusing (:

**Pull request merged**